x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/transform/action/ValidateTransformAction.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.core.transform.action;
+
+import org.elasticsearch.action.ActionRequestValidationException;
+import org.elasticsearch.action.ActionResponse;
+import org.elasticsearch.action.ActionType;
+import org.elasticsearch.action.support.master.AcknowledgedRequest;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.xpack.core.common.validation.SourceDestValidator;
+import org.elasticsearch.xpack.core.transform.transforms.TransformConfig;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.Objects;
+
+public class ValidateTransformAction extends ActionType<ValidateTransformAction.Response> {
+
+    public static final ValidateTransformAction INSTANCE = new ValidateTransformAction();
+    public static final String NAME = "cluster:admin/transform/validate";
+
+    private ValidateTransformAction() {
+        super(NAME, ValidateTransformAction.Response::new);
+    }
+
+    public static class Request extends AcknowledgedRequest<Request> {
+
+        private final TransformConfig config;
+        private final boolean deferValidation;
```

Comment:
I see no harm having `_validate` as REST endpoint. It might be useful for the UI, calling the lightweight `_validate` instead of `_preview` after the data part has been finished.

Reply:
Let's postpone making `_validate` a REST endpoint. Although I agree it will be useful in the UI eventually, it's not going to be used in `7.13`.